test/opt/ir_builder.cpp

```diff
@@ -296,6 +296,96 @@ OpFunctionEnd
   Match(text, context.get());
 }
 
+TEST_F(IRBuilderTest, ConstantAdder) {
+  const std::string text = R"(
+; CHECK: [[uint:%\w+]] = OpTypeInt 32 0
+; CHECK: [[u0:%\w+]] = OpConstant [[uint]] 13
+; CHECK: [[sint:%\w+]] = OpTypeInt 32 1
+; CHECK: [[u1:%\w+]] = OpConstant [[sint]] -1
+; CHECK: [[u2:%\w+]] = OpConstant [[uint]] 1
+; CHECK: [[u3:%\w+]] = OpConstant [[sint]] 34
+; CHECK: [[u2:%\w+]] = OpConstant [[uint]] 0
+; CHECK: [[u3:%\w+]] = OpConstant [[sint]] 0
+OpCapability Shader
+OpCapability Linkage
+OpMemoryModel Logical GLSL450
+%1 = OpTypeVoid
+%2 = OpTypeFunction %1
+%3 = OpFunction %1 None %2
+%4 = OpLabel
+OpReturn
+OpFunctionEnd
+)";
+
+  std::unique_ptr<ir::IRContext> context =
+      BuildModule(SPV_ENV_UNIVERSAL_1_2, nullptr, text);
+  EXPECT_NE(nullptr, context);
+
+  opt::InstructionBuilder builder(
+      context.get(), &*context->module()->begin()->begin()->begin());
+  EXPECT_NE(nullptr, builder.Add32BitUnsignedIntegerConstant(13));
+  EXPECT_NE(nullptr, builder.Add32BitSignedIntegerConstant(-1));
+
+  // Try adding the same constants again to make sure they aren't added.
```

Comment:
The check statements are insufficient for this. Effcee does not match full lines so `%uint_13 = OpConstant %uint 13` would match `OpConstant %uint 1`. You should either save the results of the first call and compare them to the second call or you could use some `CHECK-NOT` commands. Just be aware that the negative pattern is only ensured between two matches.